tableauserverclient/server/query.py

```diff
@@ -15,11 +16,54 @@ def __init__(self, model):
         self._pagination_item = None
 
     def __iter__(self):
-        self._fetch_all()
-        return iter(self._result_cache)
+        self.request_options.pagenumber = 1
+        self._result_cache = None
+        total = self.total_available
+        size = self.page_size
+        yield from self._result_cache
+        for page in range(1, math.ceil(total / size)):
+            self.request_options.pagenumber = page + 1
+            self._result_cache = None
+            self._fetch_all()
+            yield from self._result_cache
 
     def __getitem__(self, k):
-        return list(self)[k]
+        page = self.page_number
+        size = self.page_size
+
+        page_range = range((page - 1) * size, page * size)
+
```

Comment:
could you add some comments in here, or maybe rename the index variables?  it's a bit of an unexpected wall of text, I think some framing would make it easier to understand.

Reply:
Not gonna hold up the whole thing for that though, it can be a separate item.